src/components/Drawer/index.js

```diff
@@ -0,0 +1,95 @@
+import React from "react";
+import PropTypes from "prop-types";
+import classNames from "classnames";
+
+import { Consumer } from "./Context";
+import {
+  DrawerContainer,
+  DrawerContent,
+  HeaderContent,
+  CloseButton
+} from "./Drawer.styles";
+import Gradient from "../Gradient";
+import Text from "../Text/Base";
+import HamburgerIcon from "../Icons/HamburgerIcon";
+
+export default class Drawer extends React.Component {
+  static propTypes = {
+    children: PropTypes.oneOfType([
+      PropTypes.node,
+      PropTypes.element,
+      PropTypes.func
+    ]),
+    header: PropTypes.oneOfType([
+      PropTypes.node,
+      PropTypes.func,
+      PropTypes.string
+    ])
+  };
+
+  static defaultProps = { children: null, header: null };
+
+  renderChildren = props => {
+    const { children } = this.props;
+    if (typeof children === "function") return children(props);
+
+    return children;
+  };
+
+  renderHeader = ({ toggleDrawer, isOpen }) => {
+    const { header } = this.props;
+
+    if (typeof header === "function") return header({ toggleDrawer, isOpen });
+
+    return (
+      <Gradient
+        className="gradient--spotlight"
+        style={{ height: "60px", position: "relative" }}
+      >
+        <HeaderContent>
+          <div>
+            {typeof content === "string" ? (
+              <Text size="kilo" variant="light" primary weight="semiBold">
+                {header}
+              </Text>
+            ) : (
+              header
+            )}
+          </div>
+          <CloseButton type="button" onClick={toggleDrawer}>
+            <HamburgerIcon
+              className={classNames({
+                hamburger: true,
+                "hamburger--opened": isOpen,
+                "hamburger--closed": !isOpen
+              })}
+            />
+          </CloseButton>
+        </HeaderContent>
+      </Gradient>
+    );
+  };
+
+  render() {
```

Comment:
Will this component be exported as a standalone? If so, do we want to support the passing of additional props to this component as well as the renderHeader/renderChildren methods?